core/src/main/java/org/apache/gravitino/credential/CatalogCredentialManager.java

```diff
@@ -51,6 +51,16 @@ public Credential getCredential(String credentialType, CredentialContext context
     return credentialCache.getCredential(credentialCacheKey, cacheKey -> doGetCredential(cacheKey));
   }
 
+  // Get credential with only one credential provider.
+  public Credential getCredential(CredentialContext context) {
+    if (credentialProviders.size() == 0) {
+      throw new RuntimeException("There are not any credential provider for the catalog.");
+    } else if (credentialProviders.size() > 1) {
+      throw new RuntimeException("There are multiple credential providers for the catalog.");
+    }
+    return getCredential(credentialProviders.keySet().iterator().next(), context);
```

Comment:
Can we use a more meaningful exception instead of `RuntimeException`?

Reply:
ok

Reply:
updated